File: retrofit/src/main/java/com/example/retrofit/MainActivity.java
package com.example.retrofit;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.example.retrofit.network.NetworkHelper;
import com.example.retrofit.network.response.VideoListResponse;
import com.google.gson.Gson;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;
import io.reactivex.functions.Consumer;

public class MainActivity extends AppCompatActivity {

    private CompositeDisposable mDisposable = new CompositeDisposable();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Disposable disposable = NetworkHelper.getInstance().getVideoList("", "", "")
                .subscribe(new Consumer<VideoListResponse>() {
                    @Override
                    public void accept(VideoListResponse videoListResponse) throws Exception {

                    }
                }, new Consumer<Throwable>() {
                    @Override
                    public void accept(Throwable throwable) throws Exception {

                    }
                });
        mDisposable.add(disposable);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mDisposable.dispose();
    }
}

File: tinkerfix/src/main/java/com/konstant/tinkerfix/MainActivity.kt
package com.konstant.tinkerfix

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.tencent.tinker.lib.tinker.TinkerInstaller

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        TinkerInstaller.onReceiveUpgradePatch(getApplicationContext(), "");
    }
}